alerting/src/main/kotlin/org/opensearch/alerting/util/destinationmigration/DestinationMigrationUtilService.kt

```diff
@@ -64,17 +64,25 @@ class DestinationMigrationUtilService {
             try {
                 runningLock = true
 
-                val destinationsToMigrate = retrieveDestinationsToMigrate(client)
-                logger.info("Need to migrate ${destinationsToMigrate.size} destinations")
-                if (destinationsToMigrate.isEmpty()) {
+                val emailAccountsToMigrate = retrieveConfigsToMigrate(client, "email_account")
```

Comment:
Can you add a comment for this method if is thread safe? Is it idempotent? Can this be scheduled on other nodes of the cluster?

Reply:
Yeah, if you look a few lines above, you'll see a `runningLock` with a comment. That's used to ensure that only a single instance of `migrateDestinations()` is ever running.

The `DestinationMigrationCoordinator` is what schedules this job and it uses a similar mechanism (and is also only scheduled by the elected cluster manager node)